x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java

```diff
@@ -24,163 +24,24 @@
-
-    /**
-     * Base64 encoding and decoding functions.
-     */
-    public static final NodeFeature BASE64_DECODE_ENCODE = new NodeFeature("esql.base64_decode_encode", true);
-
-    /**
-     * Support for the :: casting operator
-     */
-    public static final NodeFeature CASTING_OPERATOR = new NodeFeature("esql.casting_operator", true);
-
-    /**
-     * Blocks can be labelled with {@link org.elasticsearch.compute.data.Block.MvOrdering#SORTED_ASCENDING} for optimizations.
-     */
-    public static final NodeFeature MV_ORDERING_SORTED_ASCENDING = new NodeFeature("esql.mv_ordering_sorted_ascending", true);
-
-    /**
-     * Support for metrics counter fields
-     */
-    public static final NodeFeature METRICS_COUNTER_FIELDS = new NodeFeature("esql.metrics_counter_fields", true);
-
-    /**
-     * Cast string literals to a desired data type for IN predicate and more types for BinaryComparison.
-     */
-    public static final NodeFeature STRING_LITERAL_AUTO_CASTING_EXTENDED = new NodeFeature(
-        "esql.string_literal_auto_casting_extended",
-        true
-    );
-
-    /**
-     * Support for metadata fields.
-     */
-    public static final NodeFeature METADATA_FIELDS = new NodeFeature("esql.metadata_fields", true);
-
-    /**
-     * Support for timespan units abbreviations
-     */
-    public static final NodeFeature TIMESPAN_ABBREVIATIONS = new NodeFeature("esql.timespan_abbreviations", true);
-
-    /**
-     * Support metrics counter types
-     */
-    public static final NodeFeature COUNTER_TYPES = new NodeFeature("esql.counter_types", true);
-
     /**
      * Support metrics syntax
      */
     public static final NodeFeature METRICS_SYNTAX = new NodeFeature("esql.metrics_syntax");
 
-    /**
-     * Internal resolve_fields API for ES|QL
-     */
-    public static final NodeFeature RESOLVE_FIELDS_API = new NodeFeature("esql.resolve_fields_api", true);
-
     private Set<NodeFeature> snapshotBuildFeatures() {
         assert Build.current().isSnapshot() : Build.current();
         return Set.of(METRICS_SYNTAX);
     }
 
     @Override
     public Set<NodeFeature> getFeatures() {
-        Set<NodeFeature> features = Set.of(
-            ASYNC_QUERY,
-            AGG_VALUES,
-            BASE64_DECODE_ENCODE,
-            MV_SORT,
-            DISABLE_NULLABLE_OPTS,
-            ST_X_Y,
-            FROM_OPTIONS,
-            SPATIAL_POINTS_FROM_SOURCE,
-            SPATIAL_SHAPES,
-            ST_CENTROID_AGG,
-            ST_INTERSECTS,
-            ST_CONTAINS_WITHIN,
-            ST_DISJOINT,
-            STRING_LITERAL_AUTO_CASTING,
-            CASTING_OPERATOR,
-            MV_ORDERING_SORTED_ASCENDING,
-            METRICS_COUNTER_FIELDS,
-            STRING_LITERAL_AUTO_CASTING_EXTENDED,
-            METADATA_FIELDS,
-            TIMESPAN_ABBREVIATIONS,
-            COUNTER_TYPES,
-            RESOLVE_FIELDS_API
-        );
+        Set<NodeFeature> features = Set.of();
```

Comment:
This is always empty set now, below could be simplified to not union with empty.